lib/ansible/modules/files/file.py

```diff
@@ -530,6 +530,11 @@ def execute_touch(path, follow, timestamps):
 
         result['changed'] = changed
         result['diff'] = diff
+    else:
+        diff = initial_diff(path, 'touch', prev_state)
+        # Even if check_mode = True, touch always changes the file because it updates its timestamp
+        result['changed'] = True
```

Comment:
If the `access_time` and `modification_time` options are set to "preserve" check mode is still going to show changed=True, even though no change will be made without check mode.